### PR TITLE
KV specific LF library

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by cocreature** [comment]
You either branched off from the wrong commit or you are targetting the wrong branch.

**Comment by remyhaemmerle-da** [comment]
> You either branched off from the wrong commit or you are targetting the wrong branch.

You are two fast. I did not have time to fix the branch before you react. 
You probably need to introduce a 5min delay in your notification.

It is fixed now.

**Review by cocreature — COMMENTED** [review]
lgtm, can we merge this in a standalone PR into main, perhaps also with standalone tests and then rebase the other branch on top?

**Comment by remyhaemmerle-da** [comment]
> lgtm, can we merge this in a standalone PR into main, perhaps also with standalone tests and then rebase the other branch on top?

Sound a good idea to merge it as a standalone.
I will sync with @miklos-da  tomorow on how we proceed.

File: daml-lf/kvutils/BUILD.bazel
Comment (remyhaemmerle-da) [inline daml-lf/kvutils/BUILD.bazel:13]: Do we need to publish the jar on maven ? 
Is there is a way to ship in the jar `//ledger/participant-state/kvutils:kvutils` ?

Comment (cocreature) [reply daml-lf/kvutils/BUILD.bazel:13]: not in any nice way. If it’s a separate bazel target it’s going to be published separately.

**Review by cocreature — APPROVED** [review]
lgtm, thank you!

File: daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala
Comment (cocreature) [inline daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala:33]: needs a `.view.filterKeys(…).toMap` on Scala 2.13 (I tghink the `toMap` is optional if you’re fine with the lazy view.

File: daml-lf/kvutils/BUILD.bazel
Comment (remyhaemmerle-da) [reply daml-lf/kvutils/BUILD.bazel:13]: I add the maven tag just below and the entry in `release/artifacts.yaml`. Do I miss something ?

Comment (cocreature) [reply daml-lf/kvutils/BUILD.bazel:13]: that should be sufficient. In general, CI should yell at you if you forget an entry.

Comment (remyhaemmerle-da) [reply daml-lf/kvutils/BUILD.bazel:13]: :+1:

File: daml-lf/kvutils/BUILD.bazel
Comment (miklos-da) [inline daml-lf/kvutils/BUILD.bazel:1]: Can you name this module as `transaction-support` or `transaction-utils`, please? I.e., I would like to avoid tying its name to kvutils as it may be confusing.

File: daml-lf/kvutils/README.md
Comment (miklos-da) [inline daml-lf/kvutils/README.md:9]: Instead of adding a note here, let's modify the `CODEOWNERS` file and add the relevant people there instead to have them automatically notified when this module is being modified.

File: daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala
Comment (miklos-da) [inline daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala:26]:
```suggestion
        // Fetch and Lookup are filtered out from roots
```

File: daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala
Comment (miklos-da) [inline daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala:29]:
```suggestion
        // Fetch and Lookup are filtered out from nodes
```

File: daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala
Comment (miklos-da) [inline daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala:36]:
```suggestion
        // Exercises are preserved but Fetch and Lookup nodes are filtered out from their children.
```

File: daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala
Comment (miklos-da) [inline daml-lf/kvutils/src/test/scala/com/daml/lf/kvutils/NormalizerSpec.scala:32]:
```suggestion
        // Create nodes are preserved.
```

**Review by miklos-da — APPROVED** [review]
LGTM as long as the module will be renamed.

File: daml-lf/kvutils/BUILD.bazel
Comment (cocreature) [reply daml-lf/kvutils/BUILD.bazel:1]: But it is tied to kvutils? Dropping fetch nodes is not valid in general.

Comment (miklos-da) [reply daml-lf/kvutils/BUILD.bazel:1]: Yes, it is tied to `kvutils`, however, that doesn't mean we need to repeat that name in a Maven module's name.

Comment (remyhaemmerle-da) [reply daml-lf/kvutils/BUILD.bazel:1]: let's go for `kv-transaction-support`

File: daml-lf/kvutils/README.md
Comment (remyhaemmerle-da) [reply daml-lf/kvutils/README.md:9]: :+1:

File: CODEOWNERS
Comment (miklos-da) [inline CODEOWNERS:45]: Can you keep these in alphabetical order, please? It will be much easier for humans to scan this list ;)

File: daml-lf/kv-transaction-support/README.md
Comment (miklos-da) [inline daml-lf/kv-transaction-support/README.md:6]:
```suggestion
should not be used outside of the KV integration kit.
```

File: daml-lf/kv-transaction-support/BUILD.bazel
Comment (miklos-da) [inline daml-lf/kv-transaction-support/BUILD.bazel:19]: To follow Maven conventions, this should rather be:
```suggestion
    tags = ["maven_coordinates=com.daml:daml-lf-kv-transaction-support:__VERSION__"],
```

Comment (remyhaemmerle-da) [reply daml-lf/kv-transaction-support/BUILD.bazel]: good catch.

File: CODEOWNERS
Comment (remyhaemmerle-da) [inline CODEOWNERS:45]: I make:
- @remyhaemmerle-da owner of all `/daml-lf`
- insert a new entry `/daml-lf/kv-transaction-support/ @cocreature @gerolf-da @miklos-da`
- sort line lexicographically 
- sort the owners lexicographically

**Pull request merged**
